istioctl/pkg/dashboard/dashboard.go

```diff
@@ -411,23 +416,27 @@ func istioDebugDashCmd(ctx cli.Context) *cobra.Command {
   istioctl d istiod-debug pilot-123-456.istio-system
 `,
 		RunE: func(c *cobra.Command, args []string) error {
-			if labelSelector == "" && len(args) < 1 {
+			if labelSelector == "" && opts.Revision == "" && len(args) < 1 {
 				c.Println(c.UsageString())
-				return fmt.Errorf("specify a pod or --selector")
+				return fmt.Errorf("specify a pod, --selector, or --revision")
 			}
 
-			if labelSelector != "" && len(args) > 0 {
+			if (labelSelector != "" && len(args) > 0) || (labelSelector != "" && opts.Revision != "") || (len(args) > 0 && opts.Revision != "") {
 				c.Println(c.UsageString())
-				return fmt.Errorf("name cannot be provided when a selector is specified")
+				return fmt.Errorf("only one of name, --selector, or --revision can be specified")
 			}
-
 			client, err := ctx.CLIClientWithRevision(opts.Revision)
 			if err != nil {
 				return fmt.Errorf("failed to create k8s client: %v", err)
 			}
 
 			var podName, ns string
+			if opts.Revision != "" {
+				labelSelector = "istio.io/rev=" + opts.Revision + ", app=istiod"
```

Comment:
Isn't this already done on line 428?

Reply:
No, this is to filter the replicas of istiod, which will not be implemented in the client.

There is similar processing logic elsewhere: https://github.com/istio/istio/blob/master/istioctl/pkg/admin/istiodconfig.go#L418-L431

Reply:
hmm, it seems like the client is a little inconsistent with respecting the`. revision` field. Setting the `revision` field kinda suggests that **only** objects with that revision will be returned. You see it here

https://github.com/istio/istio/blob/24348a28263eae7ee1167e4eb92f0dd01077efbb/pkg/kube/client.go#L927-L941

and I do wonder if we want to update `PodsForSelector` to do a similar thing. It just feels weird to set a field and never actually use it.

Reply:
Sorry for the delay. If this filtering code is adjusted, some subsequent logics will also need to be adjusted, and it does not seem to bring any additional benefits. And now similar situations are also common in other places in the istio code. So I don't think it is necessary to adjust it. What do you think?

Reply:
Ok I see what you mean. I'm ok with how it is an let the approver have the final word

Reply:
Also, if you could use `label.IoIstioRev.Name` instead